src/WiFi.cpp

```diff
@@ -148,13 +148,31 @@ static void resolve_cb(uint8_t * /* hostName */, uint32_t hostIp)
 }
 
 static void ping_cb(uint32 u32IPAddr, uint32 u32RTT, uint8 u8ErrorCode) {
-  if (u8ErrorCode == 0)
-    if (WiFi._resolve == u32IPAddr)
-      WiFi._resolve = WL_PING_SUCCESS;
-    else
-      WiFi._resolve = WL_PING_DEST_UNREACHABLE;
-  else
-    WiFi._resolve = u8ErrorCode;
+	switch(u8ErrorCode){
```

Comment:
@sandeepmistry sure, I can change it to "if/else".
For constants you mean we should use predefined constants to check result within ping_cb only or you want to use these names in the new enum wl_ping_result_t?